From: Ichthyostega <prg@ichthyostega.de>
Date: Sat, 25 Jan 2020 16:13:41 +0100
Subject: CImg -- no downloads please

We want a debian build to use the sources provided in the archive,
and not to download sourcecode silently from further sources in the net.
Note: buildservers might be shielded from the net to enforce that constraint.

As workaround, the missing content will be supplied via debian/patches
---
 CImg/Makefile | 26 +++++++++++++++++---------
 1 file changed, 17 insertions(+), 9 deletions(-)

diff --git a/CImg/Makefile b/CImg/Makefile
index f7e74da..aec1207 100644
--- a/CImg/Makefile
+++ b/CImg/Makefile
@@ -158,24 +158,32 @@ endif
 # commit 39215451d0b839b4e72d32afb15ea69701a7f4eb is CImg 2.8.0
 # commit 530715cfe254d325cf8cec345210d9935fe4cf50 is CImg 2.8.1
 # commit b29d5cb0ed4ca3d94ceae31232668f646c9e45a9 is CImg 2.8.2
-CIMGVERSION=b29d5cb0ed4ca3d94ceae31232668f646c9e45a9
 
-CImg.h: Inpaint/inpaint.h
-	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/CImg.h
+##CIMGVERSION=b29d5cb0ed4ca3d94ceae31232668f646c9e45a9
+## DEB: code in this version now provided via debian/patches
+
+##CImg.h: Inpaint/inpaint.h
+#   # No silent downloads from external sources for a Debian build please!
+#
+##	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/CImg.h
 #	patch -p0 -d. < CImg-2.0.0-anisotropic.patch
 #	patch -p0 -d. < CImg-1.7.1-omp.patch
 
-Inpaint/inpaint.h:
-	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/plugins/inpaint.h
-	patch -p0 -d. < Inpaint/inpaint.h.patch
+##Inpaint/inpaint.h:
+##	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/plugins/inpaint.h
+#
+#   DEB: the following patch is applied via debian/patches (makes cleaning simpler)
+##	patch -p0 -d. < Inpaint/inpaint.h.patch
+
+##nlmeans.h:
+##	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/plugins/nlmeans.h
 
-#nlmeans.h:
-#	curl -L -s -S -o $@ https://raw.githubusercontent.com/dtschump/CImg/$(CIMGVERSION)/plugins/nlmeans.h
 
 .PHONY: distclean
 
 distclean: clean
-	-rm CImg.h Inpaint/inpaint.h
+# ...automatically removed by the Debian patching system...
+##	-rm CImg.h Inpaint/inpaint.h
 
 #git archive --remote=git://git.code.sf.net/p/gmic/source $(CIMGVERSION):src CImg.h | tar xf -
 
